src/main/java/mekanism/common/item/interfaces/IJetpackItem.java

```diff
@@ -0,0 +1,130 @@
+import mekanism.api.text.ILangEntry;
+import mekanism.common.CommonPlayerTickHandler;
+import mekanism.common.Mekanism;
+import mekanism.common.MekanismLang;
+import mekanism.common.integration.curios.CuriosIntegration;
+import mekanism.common.util.MekanismUtils;
+import mekanism.common.util.MekanismUtils.ResourceType;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.Mth;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.phys.Vec3;
+
+public interface IJetpackItem {
+
+    boolean canUseJetpack(ItemStack stack);
+    JetpackMode getJetpackMode(ItemStack stack);
+    void useJetpackFuel(ItemStack stack);
+
+    default boolean isActive(ItemStack stack) {
+        return canUseJetpack(stack) && getJetpackMode(stack) != JetpackMode.DISABLED;
+    }
+
+    enum JetpackMode implements IIncrementalEnum<JetpackMode>, IHasTextComponent {
+        NORMAL(MekanismLang.JETPACK_NORMAL, EnumColor.DARK_GREEN, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_normal.png")),
+        HOVER(MekanismLang.JETPACK_HOVER, EnumColor.DARK_AQUA, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_hover.png")),
+        DISABLED(MekanismLang.JETPACK_DISABLED, EnumColor.DARK_RED, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_off.png"));
+
+        private static final JetpackMode[] MODES = values();
+        private final ILangEntry langEntry;
+        private final EnumColor color;
+        private final ResourceLocation hudIcon;
+
+        JetpackMode(ILangEntry langEntry, EnumColor color, ResourceLocation hudIcon) {
+            this.langEntry = langEntry;
+            this.color = color;
+            this.hudIcon = hudIcon;
+        }
+
+        @Override
+        public Component getTextComponent() {
+            return langEntry.translateColored(color);
+        }
+
+        @Nonnull
+        @Override
+        public JetpackMode byIndex(int index) {
+            return byIndexStatic(index);
+        }
+
+        public ResourceLocation getHUDIcon() {
+            return hudIcon;
+        }
+
+        public static JetpackMode byIndexStatic(int index) {
+            return MathUtils.getByIndexMod(MODES, index);
+        }
+    }
+
+    /**
+     * Gets the first found jetpack from an entity, if one is worn.
+     * <br>
+     * If Curios is loaded, the curio slots will be checked as well.
+     *
+     * @param entity the entity on which to look for the jetpack
+     *
+     * @return the jetpack stack if present, otherwise an empty stack
+     */
+    @Nonnull
+    static ItemStack getActiveJetpack(LivingEntity entity) {
+        ItemStack chest = entity.getItemBySlot(EquipmentSlot.CHEST);
+        if (chest.getItem() instanceof IJetpackItem jetpackItem && jetpackItem.isActive(chest)) {
```

Comment:
So I know it is different than it was before. My thought was that if you ran out of fuel on the current one it should switch to a backup in curios. But you're right that this would be unintuitive with disabling the worn one. Maybe we switch isActive to not check the mode?

Reply:
Personally I think it shouldn't use curios at all if the mekasuit is being worn (nor should the mekasuit work in curios), making the normal jetpack fallback to curios if the one in the chestplate is empty seems like it may be reasonable though.

Reply:
It's less likely for the mekasuit to run out of fuel, but I don't know why we'd prevent it from defaulting to a curio jetpack if it did. 
I feel like we shouldn't be doing anything different for IJetpackItems though, so we'd have to put the check for whether the mekasuit is in curios in the mekasuit implementation I think, and set isActive to always true so that it won't go look at curios if that's what you decide.